src/net.c

```diff
@@ -1143,11 +1166,131 @@ coap_send(coap_session_t *session, coap_pdu_t *pdu) {
     memset(&block, 0, sizeof(block));
   }
 
+#if COAP_Q_BLOCK_SUPPORT
+  /* Indicate support for Q-Block if appropriate */
+  if (session->block_mode & COAP_BLOCK_TRY_Q_BLOCK &&
+      session->type == COAP_SESSION_TYPE_CLIENT &&
+      COAP_PDU_IS_REQUEST(pdu)) {
+    /*
+     * When the pass / fail response for Q-Block is received, this PDU will
+     * get transmitted.
+     */
+    return coap_block_test_q_block(session, pdu);
+  }
+#endif /* COAP_Q_BLOCK_SUPPORT */
+
+#if !defined(WITH_LWIP)
+  unsigned ref;
+
+  /* TODO for LwIP */
+  /*
+   * Need to wait for Q-Block test to get out and response back before
+   * continuing
+   */
+  ref = ++session->ref;
+  while (session->saved_pdu) {
+    int result = coap_io_process(session->context, 1000);
```

Comment:
I wonder about this `1000`. Do we want to define a pre-processor symbol for this?

Reply:
I have rewritten this logic (which covers previous comment), modeled on how the Extended Token test is done (which work was done after doing this code) to re-do the 'does the remote support Q-Block' test. Code is now simplified.

Coding pushed as a separate commit to make checking the changes easier.  This separate commit will get squashed before pushing the PR into develop.